packages/studio-plugin/src/createStudioPlugin.ts

```diff
@@ -33,6 +35,24 @@ export default async function createStudioPlugin(
   const pathToUserProjectRoot = getProjectRoot(cliArgs);
 
   const studioConfig = await getStudioConfig(pathToUserProjectRoot, cliArgs);
+
+  let pagesDevelopmentPort: Promise<number | undefined>;
+  if (studioConfig.isPagesJSRepo) {
+    pagesDevelopmentPort = startPagesDevelopmentServer();
+    pagesDevelopmentPort
+      .then((port) => {
+        if (port) {
+          process.on("exit", () => {
+            console.log("Studio is shutting down");
+            execSync(`npx kill-port ${port}`);
+          });
+        }
+      })
+      .catch((e) => {
```

Comment:
The .catch prevents unhandled rejections which is nice! But I think I'd still like to await the promise itself at the end just to avoid potential weird edge cases with studio starting before `pages dev` is done. If we await the promise I think we don't need this .catch anymore.